app/gosqs/send_message_test.go

```diff
@@ -0,0 +1,111 @@
+package gosqs
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/Admiral-Piett/goaws/app"
+	"github.com/Admiral-Piett/goaws/app/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSendMessage_MaximumMessageSize_Success(t *testing.T) {
+	req, err := http.NewRequest("POST", "/", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	app.SyncQueues.Queues["test_max_message_size"] =
+		&app.Queue{Name: "test_max_message_size", MaximumMessageSize: 100}
+
+	form := url.Values{}
+	form.Add("Action", "SendMessage")
+	form.Add("QueueUrl", "http://localhost:4100/queue/test_max_message_size")
+	form.Add("MessageBody", "test%20message%20body%201")
+	form.Add("Version", "2012-11-05")
+	req.PostForm = form
+
+	status, response := SendMessageV1(req)
+
+	// Check the status code is what we expect.
+	if status != http.StatusOK {
+		t.Errorf("handler returned wrong status code: got %v want %v",
+			status, http.StatusOK)
+	}
+
+	// Check the response body is what we expect.
+	sendMessageResponse, ok := response.(models.SendMessageResponse)
+	assert.True(t, ok)
+	if len(sendMessageResponse.Result.MD5OfMessageBody) == 0 {
+		t.Errorf("handler returned unexpected body: got %v",
```

Comment:
Let's use `assert` and check the entire message body, at least just once.

Reply:
Also, getting a good `FIFO` sequence increment?  At least in one of these?

Reply:
Updated!